.github/actions/start-opensearch-with-one-plugin/action.yml

```diff
@@ -59,41 +58,38 @@ runs:
       run: mv ./build/distributions/${{ inputs.plugin-name }}-*.zip ${{ inputs.plugin-name }}.zip
       shell: bash
 
-    # Install the plugin, runs its start-script, and start the OpenSearch server
+    # Install the plugin
     - name: Install Plugin into OpenSearch for Linux
       if: ${{ runner.os == 'Linux'}}
       run: |
-        cat > os-ep.sh <<EOF
-        yes | opensearch-plugin install file:///docker-host/${{ inputs.docker-host-plugin-zip }}.zip
-        chmod +x plugins/${{ inputs.plugin-name }}/tools/${{ inputs.plugin-start-script }}.sh
-        yes | plugins/${{ inputs.plugin-name }}/tools/${{ inputs.plugin-start-script }}.sh
-        chown 1001:1001 -R /opensearch
-        su -c "/opensearch/bin/opensearch" -s /bin/bash opensearch
-        EOF
-        docker build -t opensearch-test -f- . <<EOF
-        FROM ubuntu:latest
-        COPY --chown=1001:1001 os-ep.sh /docker-host/
-        COPY --chown=1001:1001 ${{ inputs.plugin-name }}.zip /docker-host/${{ inputs.docker-host-plugin-zip }}.zip
-        COPY --chown=1001:1001 opensearch* /opensearch/
-        RUN chmod +x /docker-host/os-ep.sh
-        RUN useradd -u 1001 -s /sbin/nologin opensearch
-        ENV PATH="/opensearch/bin:${PATH}"
-        WORKDIR /opensearch/
-        ENTRYPOINT /docker-host/os-ep.sh
-        EOF
+        chmod +x ./opensearch-3.0.0-SNAPSHOT/bin/opensearch-plugin
+        /bin/bash -c "yes | ./opensearch-3.0.0-SNAPSHOT/bin/opensearch-plugin install file:$(pwd)/opensearch-security.zip"
       shell: bash
 
     - name: Install Plugin into OpenSearch for Windows
       if: ${{ runner.os == 'Windows'}}
       run: |
-        'y' | .\opensearch-${{ inputs.opensearch-version }}-SNAPSHOT\bin\opensearch-plugin.bat install file:$pwd\${{ inputs.plugin-name }}.zip
-        'y', 'y', 'N' | .\opensearch-${{ inputs.opensearch-version }}-SNAPSHOT\plugins\${{ inputs.plugin-name }}\tools\${{ inputs.plugin-start-script }}.bat
+        'y' | .\opensearch-${{ inputs.opensearch-version }}-SNAPSHOT\bin\opensearch-plugin.bat install file:$(pwd)\${{ inputs.plugin-name }}.zip
+      shell: pwsh
+
+    # Run any configuration scripts
+    - name: Run Setup Script for Linux
+      if: ${{ runner.os == 'Linux' && inputs.setup-script != '' }}
+      run: | 
+        echo "running linux setup"
+        chmod +x ./setup.sh 
+        ./setup.sh
+      shell: bash
+
+    - name: Run Setup Script for Windows
+      if: ${{ runner.os == 'Windows' && inputs.setup-script != '' }}
+      run: .\setup.bat
       shell: pwsh
 
     # Run OpenSearch
     - name: Run OpenSearch with plugin on Linux
       if: ${{ runner.os == 'Linux'}}
-      run: docker run --name ops -d -p 9200:9200 -p 9600:9600 -i opensearch-test:latest
+      run: /bin/bash -c "./opensearch-${{ inputs.opensearch-version }}-SNAPSHOT/bin/opensearch &"
```

Comment:
This command is set to run in background here with `&`. Any specific reason why background and not normally?

Reply:
Because you want it to be a background so that it is able run detached outside of docker.

Reply:
okie...in my understanding... everything runs outside docker now..so I'm still not sure how this would help (not saying it shouldn't be as it is)

Reply:
Basically, the process fails if you do not detach the call. So by making it a background process everything works because then you can have it operate in the background and then go make a curl to it. Otherwise you cannot/would need to open a new terminal during the execution to curl to it. This is my understanding at least--it fails without the & so I added it at Craig's suggestion.

Reply:
@DarshitChanpura This call is blocking if it is not detached and the rest of the action does not execute.

Reply:
@scrawfor99 So is there any difference between running it in this way and running it with `docker run`?